### PR TITLE
Add Source interface proposal

### PR DESCRIPTION
**Pull request by pdambrauskas**
@Minutis what do you think about this interface? can you see something missing?

### CODE REVIEW

**Comment by pdambrauskas** [comment]
I'd disagree. Batching strategies are strongly dependent on source/sink specification. Implementation should handle opening DB connections/files, commiting transactions or MQ offsets and etc.
Those cases would be more complicated. We would not have controll of the batch in one methods' call
IMO this batch isolation is superior to processing record-by-record

File: racoon-commons/src/main/java/run/racoon/commons/domain/PullContext.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package run.racoon.commons.domain;
+
+public class PullContext {
```
Comment (Minutis) [inline racoon-commons/src/main/java/run/racoon/commons/domain/PullContext.java]: This is not necessary.

**Comment by pdambrauskas** [comment]
@Minutis, I've added example implementation of jdbc source task. We need to add more implementations, to see what is needed and stabilize the interface. Can you review this PR, so that we can move with example implementations and see what is missing on the way?

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package run.racoon.commons.domain;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public class Record {
+    private final Set<Cell> cells;
+    // TODO: Do we need offset there?
```
Comment (Minutis) [inline racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: Let's leave it here for now.

**Pull request merged**